devito/dimension.py

```diff
@@ -28,6 +29,16 @@ def get_varname(self):
         self._count += 1
         return name
 
+    @property
+    def ccode(self):
+        """C-level variable name of this dimension"""
+        return "%s_size" % self.name if self.size is None else "%d" % self.size
```

Comment:
shall we establish a convention for all methods that produce C code ? as in, calling them ``cname``, ``cdecl``, ``...``

This actually makes me think that we might benefit from a common superclass that represent "Objects that can produce c code"